chemprop/data/utils.py

```diff
@@ -16,16 +16,31 @@
 from chemprop.features import load_features, load_valid_atom_features
 
 
-def preprocess_smiles_columns(smiles_columns: Optional[Union[str, List[Optional[str]]]]) -> List[Optional[str]]:
+def preprocess_smiles_columns(path: str,
+                              smiles_columns: Optional[Union[str, List[Optional[str]]]],
+                              number_of_molecules: int = 1) -> List[Optional[str]]:
     """
-    Preprocesses the :code:`smiles_columns` variable to ensure that it is a list.
+    Preprocesses the :code:`smiles_columns` variable to ensure that it is a list of column
+    headings corresponding to the columns in the data file holding SMILES.
 
+    :param path: Path to a CSV file.
     :param smiles_columns: The names of the columns containing SMILES.
                            By default, uses the first :code:`number_of_molecules` columns.
+    :param number_of_molecules: The number of molecules with associated SMILES for each
+                           data point.
     :return: The preprocessed version of :code:`smiles_columns` which is guaranteed to be a list.
     """
-    smiles_columns = smiles_columns if smiles_columns is not None else [None]
-    smiles_columns = [smiles_columns] if type(smiles_columns) != list else smiles_columns
+    if type(smiles_columns) != list:
```

Comment:
Type checking should be done using `not isinstance(smiles_columns, list)`.

Reply:
Thanks! There's a lot of python best practices that I need to pick up. Please continue to note them whenever you see them and hopefully I'll have it right for next time.